submissions/Dovahkiin1991/friends-app/app.js

```diff
@@ -0,0 +1,339 @@
+                    <p><strong>User Location:</strong></p>
+                    <p>${new Date(user.dob.date).toLocaleDateString('en-us', { weekday:"long", year:"numeric", month:"short", day:"numeric"}) }</p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Location:</strong></p>
+                    <p>${user.location.country}</p>
+                    <p>${user.location.state}</p>
+                    <p>${user.location.city}</p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Contacts:</strong></p>
+                    <p>
+                        <a href="mailto:${user.email}">${user.email}</a>
+                    </p>
+                    <p>
+                        <a href="tel:${user.phone}">${user.phone}</a>
+                    </p>
+                    <p>&nbsp;</p>
+                    <p><strong>User Gender:</strong></p>
+                    <p class="uppercase">${user.gender}</p>
+                </div>
+
+                <div class="user-button user-button--close">
+                    Back to Friends
+                </div>
+            </div>
+        </div>
+    `;
+    mainContainer.innerHTML = singleUser;
+}
+
+// filters
+let filter = document.querySelector('.filter__block input');
+nameInput.addEventListener('input', (e) => {
+    const value = e.preventDefault();
+    const pipes = [filterByName];
+    
+    if (getOpt('alphabet')) {
+        pipes.push(sortByAlphabet);
+    }
+    if (getOpt('age')) {
+        pipes.push(sortByAge);
+    }
+    if (getOpt('gender')) {
+        pipes.push(filterByGender);
+    }
+
+    renderUsers(pipes);
+});
+
+document.querySelectorAll('.filter__block input[type="radio"]').forEach((elem) => {
+    elem.addEventListener("change", function(event) {
+
+        switch (event.target.name) {
+            case 'alphabet':
+                uncheckRadio('age');
+                renderUsers([
+                    filterByName,
+                    filterByGender,
+                    sortByAlphabet
+                ]);
+
+                break;
+            case 'gender':
+                const pipes = [
+                    filterByName,
+                    filterByGender,
+                ];
+                if (getOpt('alphabet')) {
+                    pipes.push(sortByAlphabet);
+                }
+                if (getOpt('age')) {
+                    pipes.push(sortByAge);
+                }
+                renderUsers(pipes);
+
+                break;
+            case 'age':
+                uncheckRadio('alphabet');
+                renderUsers([
+                    filterByName,
+                    filterByGender,
+                    sortByAge
+                ]);
+                break;
+            default:
+                // renderUsers();
+        }
+    });
+});
+
+const filterByName = (user) => {
+    const searchValue = nameInput.value.trim().toLowerCase();
+    return user.name.first.toLowerCase().includes(searchValue);
+}
```

Comment:
When you read someone's code and see `fitlerByName` as a function name, what do you expect from such a function? Usually, such a function would do the following:
1. Get some kind of array that should be filtered as a parameter.
2. Get a name to filter by as a parameter.
3. Make some action to produce a filtered array.
4. Return this filtered array.

Let's see what your function does:
1. Gets user as a parameter.
2. Gets some name out of an HTML element somewhere in the function's scope (that potentially could be absent).
3. Makes some actions with this name.
4. Checks if provided user name includes the produced value.
5. Returns a boolean result.

Do you see where I'm going with this?